bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/MongoCryptImpl.java

```diff
@@ -217,7 +235,7 @@ public MongoCryptContext createDataKeyContext(final String kmsProvider, final Mo
         } else if (kmsProvider.equals("local")) {
             success = mongocrypt_ctx_setopt_masterkey_local(context);
         } else {
-            throw new IllegalArgumentException("Unsupported KMS provider " + kmsProvider);
+            success = mongocrypt_ctx_setopt_key_encryption_key(context, toBinary(options.getMasterKey()).getBinary());
```

Comment:
The spec says:

> This [mongocrypt_ctx_setopt_key_encryption_key] accepts a BSON document exactly matching the data key options, but expects the BSON document to also have a "provider" field set to either "azure" or "gcp". 

I don't see where the provider field has been added.

Reply:
+1

Reply:
That was a little unclear to me as what we expect from the user, we seem to use the following format in the database:

```
"masterKey": {
    "provider": "azure",
    "keyVaultEndpoint": "my-key-vault.vault.azure.net",
    "keyName": "test-key"
}
```

And I had some expectations that the field was explicit.

I was expecting users to use that format eg:

```
{
"azure": {
    "provider": "azure",
    "keyVaultEndpoint": "my-key-vault.vault.azure.net",
    "keyName": "test-key"
}}
```

I'll update and treat the provider as implicit if not explicitly provided.

```
"azure": {
    "keyVaultEndpoint": "my-key-vault.vault.azure.net",
    "keyName": "test-key"
}
```

Reply:
Updated

Reply:
@kevinAlbs how should this be tested?

Reply:
libmongocrypt verifies that "provider" is present. If the driver omits it, the prose tests for creating a data key for Azure or GCP will fail.